distribution/tools/entitlement-runtime/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserFailureTests.java

```diff
@@ -0,0 +1,149 @@
+                    new ByteArrayInputStream(Strings.toString(builder).getBytes(StandardCharsets.UTF_8)),
+                    "test-failure-policy.yaml"
+                ).parsePolicy()
+            );
+            assertEquals(
+                "[4:5] policy parsing error for [test-failure-policy.yaml] in scope [entitlement-module-1]: "
+                    + "unknown entitlement type [does_not_exist]",
+                ppe.getMessage()
+            );
+        }
+    }
+
+    public void testEntitlementMissingParameter() throws IOException {
+        try (XContentBuilder builder = YamlXContent.contentBuilder()) {
+            builder.startObject();
+            builder.startObject("entitlement-module-1");
+            builder.startArray("entitlements");
+            builder.startObject();
+            builder.startObject("file");
+            builder.endObject();
+            builder.endObject();
+            builder.endArray();
+            builder.endObject();
+            builder.endObject();
+            PolicyParserException ppe = expectThrows(
+                PolicyParserException.class,
+                () -> new PolicyParser(
+                    new ByteArrayInputStream(Strings.toString(builder).getBytes(StandardCharsets.UTF_8)),
+                    "test-failure-policy.yaml"
+                ).parsePolicy()
+            );
+            assertEquals(
+                "[4:12] policy parsing error for [test-failure-policy.yaml] in scope [entitlement-module-1] "
+                    + "for entitlement type [file]: missing entitlement parameter [path]",
+                ppe.getMessage()
+            );
+        }
+
+        try (XContentBuilder builder = YamlXContent.contentBuilder()) {
+            builder.startObject();
+            builder.startObject("entitlement-module-1");
+            builder.startArray("entitlements");
+            builder.startObject();
+            builder.startObject("file");
+            builder.field("path", "test-path");
+            builder.endObject();
+            builder.endObject();
+            builder.endArray();
+            builder.endObject();
+            builder.endObject();
+            PolicyParserException ppe = expectThrows(
+                PolicyParserException.class,
+                () -> new PolicyParser(
+                    new ByteArrayInputStream(Strings.toString(builder).getBytes(StandardCharsets.UTF_8)),
+                    "test-failure-policy.yaml"
+                ).parsePolicy()
+            );
+            assertEquals(
+                "[6:1] policy parsing error for [test-failure-policy.yaml] in scope [entitlement-module-1] "
+                    + "for entitlement type [file]: missing entitlement parameter [actions]",
+                ppe.getMessage()
+            );
+        }
+    }
+
+    public void testEntitlementExtraneousParameter() throws IOException {
+        try (XContentBuilder builder = YamlXContent.contentBuilder()) {
+            builder.startObject();
```

Comment:
It would be much easier to read these tests if they used multi line strings, so the file content is more understandable. I know we have used builders in tests in the past for much of xcontent, indexing, etc, but in those cases I think we care about varying the underlying xcontent type. Here we only care about yaml.

Reply:
Changed.